src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -970,8 +979,20 @@ class Simplex_tree {
     Simplex_handle simplex_one = insertion_result.first;
     bool one_is_new = insertion_result.second;
     if (!one_is_new) {
-      if (filtration(simplex_one) > filt) {
-        assign_filtration(simplex_one, filt);
+      if (!(filtration(simplex_one) <= filt)) { 
+        if constexpr (SimplexTreeOptions::is_multi_parameter){
```

Comment:
If we do nothing here when `SimplexTreeOptions::is_multi_parameter = true`, we could avoid a lot of code here.
I think we could test `if constexpr (!SimplexTreeOptions::is_multi_parameter)` upper in the method